lib/ecmascript.ts

```diff
@@ -6700,10 +6605,16 @@ export function ASCIILowercase<T extends string>(str: T): T {
   // values. For example, Turkish's "I" character was the source of a security
   // issue involving "file://" URLs. See
   // https://haacked.com/archive/2012/07/05/turkish-i-problem-and-why-you-should-care.aspx/.
-  return str.replace(/[A-Z]/g, (l) => {
-    const code = l.charCodeAt(0);
-    return String.fromCharCode(code + 0x20);
-  }) as T;
+  let lowercase = '';
```

Comment:
What's the difference in behavior between `String.prototype.lowercase` and this code?  I'm wondering if it's really needed for a polyfill.

Reply:
`toLowerCase` uses [Unicode Default case mapping](https://www.unicode.org/versions/Unicode16.0.0/core-spec/chapter-3/#G33992) and on a casual reading I cannot say with any certainty what that algorithm does. _Probably_ it produces identical results if the input is guaranteed to be ASCII, so we could get rid of this, but let's save that for when we detach from usptream.